network/src/topology.rs

```diff
@@ -0,0 +1,492 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+// Network crawler:
+// Start a crawler task (similar to the peers task) which updates state. Only one peer would be
+// connected at a time to start and would be queried for peers. It would then select on peer at
+// random to continue the crawl.
+//
+// Q: extend the network protocol to include statistics or node metadata?
+// Q: when to perform centrality computation?
+
+use std::{
+    cmp::Ordering,
+    collections::{BTreeMap, HashSet},
+    hash::{Hash, Hasher},
+    net::SocketAddr,
+    ops::Sub,
+};
+
+use chrono::{DateTime, Utc};
+use nalgebra::{DMatrix, DVector, SymmetricEigen};
+use parking_lot::RwLock;
+
+// Purges connections that haven't been seen within this time (in hours).
+const STALE_CUTOFF: i64 = 4;
```

Comment:
```suggestion
const STALE_PEER_CUTOFF_TIME: i64 = 4;
```
or something similar that's a bit more informative